### PR TITLE
[DOCS] Update Spark Code Examples in Connect to In-memory Source Data

### PR DESCRIPTION
**Pull request by kwcanuck**
In [this Slack conversation](https://greatexpectationslabs.slack.com/archives/C03B8DZCJ07/p1691585817623689), @Kilo59 indicated that the code examples for Spark in the topic _[Connect to in-memory source data](https://docs.greatexpectations.io/docs/guides/connecting_to_your_data/fluent/in_memory/connect_in_memory_data/#read-your-source-data-into-a-spark-dataframe)_ were incomplete or inaccurate. This PR updates the code examples.

The JIRA ticket for this is: [DOC-586](https://greatexpectations.atlassian.net/browse/DOC-586).

## Definition of done
- [X] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [X] _Appropriate_ tests and docs have been updated

[DOC-586]: https://greatexpectations.atlassian.net/browse/DOC-586?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tests/integration/docusaurus/connecting_to_your_data/fluent_datasources/how_to_connect_to_in_memory_data_using_spark.py [hunk 1]
```diff
@@ -36,7 +35,7 @@
 
 # Python
 # <snippet name="tests/integration/docusaurus/connecting_to_your_data/fluent_datasources/how_to_connect_to_in_memory_data_using_spark.py data_asset">
-data_asset = datasource.add_dataframe_asset(name=name)
+data_asset = datasource.add_dataframe_asset(name="my_df_asset")
```
Comment (cdkini) [inline tests/integration/docusaurus/connecting_to_your_data/fluent_datasources/how_to_connect_to_in_memory_data_using_spark.py]: Could we change name on L33 instead?

File: tests/integration/docusaurus/connecting_to_your_data/fluent_datasources/how_to_connect_to_in_memory_data_using_spark.py [hunk 2]
```diff
@@ -7,6 +7,7 @@
 
 import pathlib
 import great_expectations as gx
+import pyspark.pandas as ps
```
Comment (cdkini) [inline tests/integration/docusaurus/connecting_to_your_data/fluent_datasources/how_to_connect_to_in_memory_data_using_spark.py]:
```suggestion
import os
import pyspark.pandas as ps

# Required by pyarrow>=2.0.0 within Spark to suppress UserWarning
os.environ["PYARROW_IGNORE_TIMEZONE"] = "1"
```

File: tests/integration/docusaurus/connecting_to_your_data/fluent_datasources/how_to_connect_to_in_memory_data_using_spark.py (hunk 1 above)
Comment (cdkini) [reply tests/integration/docusaurus/connecting_to_your_data/fluent_datasources/how_to_connect_to_in_memory_data_using_spark.py]: Now L40

File: tests/integration/docusaurus/connecting_to_your_data/fluent_datasources/how_to_connect_to_in_memory_data_using_spark.py (hunk 2 above)
Comment (Kilo59) [reply tests/integration/docusaurus/connecting_to_your_data/fluent_datasources/how_to_connect_to_in_memory_data_using_spark.py]: Is this going to be part of the snippet or outside of the rendered snippet? I don't think it should be part is what our users see.

Comment (cdkini) [reply tests/integration/docusaurus/connecting_to_your_data/fluent_datasources/how_to_connect_to_in_memory_data_using_spark.py]: outside the snippet

**Comment by Shinnnyshinshin** [comment]
@kwcanuck This PR is now passing tests, but needs a bit of clean up. A few more updates to come :)

**Review by Kilo59 — APPROVED** [review]

**Review by Shinnnyshinshin — APPROVED** [review]
Success! This PR can now go in. Thank you for your patience @kwcanuck :)

**Pull request merged**